server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -254,10 +257,53 @@ static private void downloadOnce(TranslogTransferManager translogTransferManager
                 location.resolve(Translog.getCommitCheckpointFileName(translogMetadata.getGeneration())),
                 location.resolve(Translog.CHECKPOINT_FILE_NAME)
             );
+        } else {
+            // When code flow reaches this block, it means we don't have any translog files uploaded to remote store.
+            // If local filesystem contains empty translog or no translog, we don't do anything.
+            // If local filesystem contains non-empty translog, we clean up these files and create empty translog.
+            logger.debug("No translog files found on remote, checking local filesystem for cleanup");
+            if (FileSystemUtils.exists(location.resolve(CHECKPOINT_FILE_NAME))) {
+                final Checkpoint checkpoint = readCheckpoint(location);
+                if (isEmptyTranslog(checkpoint) == false) {
+                    logger.debug("Translog files exist on local without any metadata in remote, cleaning up these files");
+                    // Creating empty translog will cleanup the older un-referenced tranlog files, we don't have to explicitly delete
+                    createEmptyTranslog(translogTransferManager, location, checkpoint);
+                } else {
+                    logger.debug("Empty translog on local, skipping clean-up");
+                }
+            }
         }
         logger.debug("downloadOnce execution completed");
     }
 
+    private static boolean isEmptyTranslog(Checkpoint checkpoint) {
+        return checkpoint.generation == checkpoint.minTranslogGeneration
```

Comment:
What would the value of generation be if it is empty translog?

Reply:
It could be anything. Empty translog is not necessarily on the index creation itself. Empty translog can exist for a non-active index with non-zero documents.

Reply:
tlog files without checkpoint should be cleaned up. The issue is with tlog + ckp files which can be read and considered live.